src/main/java/org/takes/http/Options.java

```diff
@@ -199,4 +186,38 @@ public long maxLatency() {
         }
         return msec;
     }
+
+    /**
+     * Convert the provided arguments into a Map. The expected
+     * format of each argument is {@code --([a-z\-]+)(=.+)?}, if any argument
+     * doesn't match with the expected format an {@code IllegalStateException}
+     * will be thrown. If no value has been assigned to a given argument, an
+     * empty {@code String} will be assigned to that parameter by default.
+     * @param args Arguments to parse.
+     * @return Map A map containing all the arguments which are actually
+     *  key/value pairs.
+     * @throws IllegalStateException If an argument doesn't match with the
+     *  expected format.
+     */
+    private static ConcurrentMap<String, String> asMap(
+        final Iterable<String> args) throws IllegalStateException {
+        final ConcurrentMap<String, String> map =
```

Comment:
@essobedo field type can be relaxed to simple `Map` as we don't use `ConcurrentMap` specific operations